ui/hooks/useIsOriginalNativeTokenSymbol.js

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useState } from 'react';
+import fetchWithCache from '../../shared/lib/fetch-with-cache';
+import { BUILT_IN_NETWORKS } from '../../shared/constants/network';
+import { DAY } from '../../shared/constants/time';
+
+export function useIsOriginalNativeTokenSymbol(chainId, ticker, type) {
+  const [isOriginalNativeSymbol, setIsOriginalNativeSymbol] = useState(null);
+
+  useEffect(() => {
+    async function getNativeTokenSymbol(networkId) {
+      try {
+        const matchedSymbolBuiltInNetwork =
+          BUILT_IN_NETWORKS[type]?.ticker ?? null;
+        if (matchedSymbolBuiltInNetwork) {
+          setIsOriginalNativeSymbol(matchedSymbolBuiltInNetwork === ticker);
+          return matchedSymbolBuiltInNetwork === ticker;
+        }
+
+        const safeChainsList = await fetchWithCache({
+          url: 'https://chainid.network/chains.json',
```

Comment:
hey @legobeat ,
thnx for your comment , the behavior is now preserved.
if the user disable the chains validation , the hook will return `true` by default and the token validation is not done.
you can check my last commit that i just pushed to understand how it's preserved.

Reply:
regarding the second part of your comment , this will be part of a long term solution.
the idea is to build a static asset of network that we maintain on our side and use it instead of calling this endpoint.
as part of the improvement, we can also make one single call on the build rather than fetching on the runtime.

now on this situation , we're allowing this fetch through the runtime only if the users enable the chainValidation toggle and that's why it's make sense for me to use a custom hook.